ee/tabby-webserver/src/service/mod.rs

```diff
@@ -215,3 +220,26 @@ pub async fn create_service_locator(
 ) -> Arc<dyn ServiceLocator> {
     Arc::new(Arc::new(ServerContext::new(logger, code).await))
 }
+
+pub fn graphql_params_to_filter(
+    after: Option<String>,
+    before: Option<String>,
+    first: Option<usize>,
+    last: Option<usize>,
+) -> Result<(Option<usize>, Option<i32>, bool), ParseIntError> {
+    match (first, last) {
+        (Some(first), None) => {
+            let after = after.map(|x| x.parse::<i32>()).transpose()?;
+            Ok((Some(first), after, false))
+        }
+        (None, Some(last)) => {
+            let before = before.map(|x| x.parse::<i32>()).transpose()?;
+            Ok((Some(last), before, true))
+        }
+        _ => Ok((None, None, false)),
+    }
+}
+
+pub fn convert_vec<A: Into<B>, B>(v: Vec<A>) -> Vec<B> {
```

Comment:
for oneliner like this, I prefer just write it down instead of reuse it as function.